src/contracts/StakedAaveV3.sol

```diff
@@ -93,17 +93,50 @@ contract StakedAaveV3 is StakedTokenV3, IStakedAaveV3 {
       amount
     );
 
-    IGhoVariableDebtTokenTransferHook cachedGhoDebtToken = ghoDebtToken;
-    if (address(cachedGhoDebtToken) != address(0)) {
-      try
-        cachedGhoDebtToken.updateDiscountDistribution(
-          from,
-          to,
-          fromBalanceBefore,
-          toBalanceBefore,
-          amount
-        )
-      {} catch (bytes memory) {}
+    address cachedGhoDebtToken = address(ghoDebtToken);
+    if (cachedGhoDebtToken != address(0)) {
+      _updateDiscountDistribution(
+        cachedGhoDebtToken,
+        from,
+        to,
+        fromBalanceBefore,
+        toBalanceBefore,
+        amount
+      );
+    }
+  }
+
+  /// @notice Assembly implementation of the gas limited call to avoid return gas bomb,
+  /// moreover call to a destructed plugin would also revert even inside try-catch block in Solidity 0.8.17
+  function _updateDiscountDistribution(
+    address cachedGhoDebtToken,
+    address from,
+    address to,
+    uint256 fromBalanceBefore,
+    uint256 toBalanceBefore,
+    uint256 amount
+  ) internal {
+    bytes4 selector = IGhoVariableDebtTokenTransferHook
+      .updateDiscountDistribution
+      .selector;
+    uint256 gasLimit = 220_000;
+    assembly ('memory-safe') {
```

Comment:
Per solidity docs, if the contract is meant to be compatible across multiple version of Solidity, it's better to use the annotation instead. Given the contract uses ^0.8.0 I think its better to apply the following suggestion:
```suggestion
    /// @solidity memory-safe-assembly
    assembly {
```